suite/tests/CMakeLists.txt

```diff
@@ -175,19 +175,30 @@ else ()
   endif (TEST_SUITE)
 endif (CALLPROF AND TEST_SUITE)
 
-# i#4059: Reduce Appveyor time by reducing the # of tests run for 32-bit.
-# This assumes that 32-bit Windows is not as important these days.
-# We remove some of the less differentiated tests.
-# We still run the full set for merges to master.
-if (WIN32 AND NOT X64 AND "$ENV{CI_TRIGGER}" STREQUAL "pull_request")
-  set(SKIP_LESS_DIFFERENTIATED_TESTS_FOR_PULL_REQUEST ON)
+if (WIN32 AND NOT X64)
+  # i#4059: Reduce testing time by reducing the # of tests run for 32-bit.
+  # This assumes that 32-bit Windows is not as important these days.
+  # We remove some of the less differentiated tests.
+  if ("$ENV{CI_TRIGGER}" STREQUAL "pull_request")
+    set(SKIP_LESS_DIFFERENTIATED_TESTS_FOR_PULL_REQUEST ON)
+  else ()
+    # i#4800,i#5873: We'd like to run the full set for merges to master but we
+    # need pre-and-post-commmit test parity.
+    set(SKIP_LESS_DIFFERENTIATED_TESTS_FOR_PULL_REQUEST ON)
```

Comment:
This is also `ON` to make our life easier in the future in case we don't want to skip these tests for a merged commit?

Reply:
Right, wanted to not give up on re-instating a longer suite in the future.